numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -3672,13 +3672,12 @@ _vec_string_with_args(PyArrayObject* char_array, PyArray_Descr* type,
         goto err;
     }
 
-    args_tuple = PyTuple_New(n);
-    if (args_tuple == NULL) {
-        goto err;
-    }
-
     while (PyArray_MultiIter_NOTDONE(in_iter)) {
         PyObject* item_result;
+        args_tuple = PyTuple_New(n);
```

Comment:
Can you move the declaration of `args_tuple` from the beginning of the function to here?